keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -0,0 +1,237 @@
+  value: string;
+  label: string | JSX.Element;
+  tooltip?: string;
+  isDisabled?: boolean;
+}
+
+interface AddUserModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  authType: string;
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>;
+  accessToken: string;
+}
+
+const roleOptions: RoleOption[] = [
+  {
+    value: "admin",
+    label: "Admin",
+    tooltip: "Admin has read/write/update/delete for every resource",
+  },
+  {
+    value: "noc",
+    label: "NOC",
+    tooltip: "NOC has the ability to view and assign alerts",
+  },
+  {
+    value: "create_new",
+    label: "Create custom role",
+    isDisabled: true,
+    tooltip: "For custom roles, contact Keep team",
+  },
+];
+
+export default function AddUserModal({
+  isOpen,
+  onClose,
+  authType,
+  setUsers,
+  accessToken,
+}: AddUserModalProps) {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [selectedRole, setSelectedRole] = useState<string>();
+  const [password, setPassword] = useState("");
+  const [addUserError, setAddUserError] = useState("");
+
+  const validateEmail = (email: string) => {
+    const re = /\S+@\S+\.\S+/;
+    return re.test(email);
+  };
+
+  const isFormValid = () => {
+    // if multi tenant, we need to validate email, else just validate password
+    if (authType === AuthenticationType.MULTI_TENANT) {
+      return validateEmail(email) && selectedRole;
+    } else {
+      return email && selectedRole && password;
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Validate form
+    if (!isFormValid()) {
+      setAddUserError("Please fill out all fields");
+      return;
+    }
+    // Validate selected role
+    if (!selectedRole) {
+      setAddUserError("Please select a role");
+      return;
+    }
+
+    // Make the API call to add the user
+    const response = await fetch(`${getApiURL()}/settings/users`, {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ email, role: selectedRole, password }),
+    });
+
+    if (response.ok) {
+      const newUser = await response.json();
+      setUsers((currentUsers) => [...currentUsers, newUser]);
+      // Reset form and close modal on successful addition
+      handleOnClose();
+    } else {
+      const errorData = await response.json();
+      setAddUserError(
+        errorData.message ||
+          errorData.detail ||
+          "Failed to add user (unknown error)"
+      );
+    }
+  };
+
+  const handleOnClose = () => {
+    onClose();
+    setEmail("");
+    setSelectedRole("");
+    setPassword("");
+    setAddUserError("");
+  };
+
+  const handleEmailChange = (email: string) => {
+    setEmail(email);
+    if (validateEmail(email)) {
+      setEmailError("");
+    } else {
+      setEmailError("Please enter a valid email address.");
+    }
+  };
```

Comment:
You can leave the email validation to the input elements. `<input type='email' />` should make sure users are passing valid emails.